### PR TITLE
lightning: replace storage.Create by storage.New (#899)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #899 to release-4.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In br repo:
git pr https://github.com/pingcap/br/pull/919
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/br.git pr/919:release-4.0-fea552d4101c
```

---

### What problem does this PR solve? 

1. `storage.Create` has been deprecated... so let's replace the remaining two calls from Lightning.
2. Lightning originally used `storage.Create(sendCreds=true)`, but 
    * the entire cloud access happens within Lightning, so we don't need to send credentials to anyone
    * this introduced `skipCheckPath=false`, which leads to #898. 

### What is changed and how it works?

Changed `storage.Create` to `storage.New` with sendCreds=false and skipCheckPath=true.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

Code changes

Side effects

Related changes

### Release Note

 - When using Lightning to import from S3, we no longer require `s3:ListBucket` permission on the entire bucket, only the data source prefix itself.

### CODE REVIEW

File: pkg/lightning/lightning.go [hunk 1]
```diff
@@ -271,7 +271,11 @@ func (l *Lightning) run(taskCtx context.Context, taskCfg *config.Config, g glue.
 	if err != nil {
 		return errors.Annotate(err, "parse backend failed")
 	}
-	s, err := storage.Create(ctx, u, true)
+	s, err := storage.New(ctx, u, &storage.ExternalStorageOptions{
+		// we skip check path in favor of delaying the error to when we actually access the file.
+		// on S3, performing "check path" requires the additional "s3:ListBucket" permission.
+		SkipCheckPath: true,
```
Comment (overvenus) [inline pkg/lightning/lightning.go:277]: What about BR? Does it need the change too?

Comment (kennytm) [reply pkg/lightning/lightning.go:277]: See #898 for details. Yes BR also needs to be fixed, but not via disabling SkipCheckPath everywhere.

**Pull request merged**